src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +136,24 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+    //this method is used to get the serverinfo object from the server
+    public ServerInfo getInfo() {
+      try (final Connection connection = this.source.connect()) {
+        Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+        if (Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+          final Uuid version = Uuid.SERIALIZER.read(connection.in());
+          return new ServerInfo(version);
+        } else {
+          // Communicate this error - the server did not respond with the type of
+          // response we expected.
+          System.out.println("Server ran into a problem retrieving the version info");
+        }
+      } catch (Exception ex) {
+          System.out.println("Connection interrupted");
+        // Communicate this error - something went wrong with the connection.
+      }
+      System.out.println("Unknown Error");
```

Comment:
It might be useful to attempt to `getMessage()` from the exception and print it. The error is unknown to us now, but at runtime we can give the user some hint as to what happened.